### DISCUSSION TITLE
Enable HTTP log to capture the requestbody and headers in the log

### DISCUSSION THREAD

**Discussion by buddhinh** [linkerd/linkerd2#13391] [Q&A]
I'm trying to enable headers and request body in linkerd-proxy HTTP log.

By default, LINKERD2_PROXY_LOG environmental variable in workload application linkerd-proxy container set as following, disabling headers and request body.

LINKERD2_PROXY_LOG:  warn,linkerd=info,hickory=error,[{headers}]=off,[{request}]=off

I have tried to enable that by changing linkerd-config configmap.

**Default parametes of linkerd-config configmap:**

logHTTPHeaders: "off"
      logLevel: warn,linkerd=info,hickory=error
      
**Applied configmap parameters**

logHTTPHeaders: "on"
      logLevel: warn,linkerd=info,hickory=error
      
However, I have failed to loh headers and request body.

Is it possible to capture the requestbody and headers? What is the correct way to do that?

**Comment by adleong** [comment]
the `logHTTPHeaders` value must be set to `insecure` to log headers in plaintext.  See: https://github.com/linkerd/linkerd2/blob/d1b2aa5dd446c1d8c3750f2ccbdc23f599c3e796/charts/linkerd-control-plane/values.yaml#L163

**Reply by buddhinh** [reply]
Thanks for the update. Request header can be captured by setting “logHTTPHeaders” to “insecure“ and log level to debug in the linkerd-config configmap.
Is there any possibility to enable logging request headers in the deployment level of the workload? (without enabling globally)

**Reply by alpeb** [reply]
Yes, you can set the annotation `config.linkerd.io/proxy-log-http-headers: insecure` in the pod template metadata of the workload.
